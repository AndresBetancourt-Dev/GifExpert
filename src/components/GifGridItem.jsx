import propTypes from "prop-types";
import React from "react";
import GifGridUser from "./GifGridUser";

const GifGridItem = ({ url, title, user }) => {
  return (
    <div className="card">
      <img src={url} alt={title} title={title} />
      <p>{title}</p>
      {user && <GifGridUser user={user} />}
    </div>
  );
};

GifGridItem.propTypes = {
  title: propTypes.string.isRequired,
  url: propTypes.string.isRequired,
  user: propTypes.object,
};

export default GifGridItem;
