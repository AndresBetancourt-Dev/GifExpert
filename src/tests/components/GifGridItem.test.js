import React from "react";
import { shallow } from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe("Test of Component : <GifGridItem/>", () => {
  const title = "One Punch Man";
  const url =
    "https://media3.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=4938fc14v1f9xsy1na3fmayewogxuckicfc7l11fgjt3yyvj&rid=giphy.gif&ct=g";
  const wrapper = shallow(<GifGridItem title={title} url={url} />);

  test("GifGridItem : Component Mounts Correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("GifGridItem : Test PropTypes", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("GifGridItem : Must Have P with the title prop value", () => {
    const p = wrapper.find("p");
    expect(p.text().trim()).toBe(title);
  });

  test("GifGridItem : Must Have Img Src and Alt with the same url prop value", () => {
    const img = wrapper.find("img");

    expect(img.prop("src")).toBe(url);
    expect(img.prop("alt")).toBe(title);
    const props = { alt: title, src: url, title };
    expect(img.props()).toEqual(props);
    expect(img.props()).toStrictEqual(props);
  });

  test("GifGridItem : Classname card", () => {
    const div = wrapper.find("div");
    const divClasses = wrapper.prop("className");
    const className = "card";

    expect(divClasses.includes(className)).toBeTruthy();
    expect(div.hasClass(className)).toBeTruthy();
  });
});
