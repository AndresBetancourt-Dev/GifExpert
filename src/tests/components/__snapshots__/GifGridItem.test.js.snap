// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test of Component : <GifGridItem/> GifGridItem : Component Mounts Correctly 1`] = `
<div
  className="card"
>
  <img
    alt="One Punch Man"
    src="https://media3.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=4938fc14v1f9xsy1na3fmayewogxuckicfc7l11fgjt3yyvj&rid=giphy.gif&ct=g"
    title="One Punch Man"
  />
  <p>
    One Punch Man
  </p>
</div>
`;

exports[`Test of Component : <GifGridItem/> GifGridItem : Test PropTypes 1`] = `
<div
  className="card"
>
  <img
    alt="One Punch Man"
    src="https://media3.giphy.com/media/yo3TC0yeHd53G/giphy.gif?cid=4938fc14v1f9xsy1na3fmayewogxuckicfc7l11fgjt3yyvj&rid=giphy.gif&ct=g"
    title="One Punch Man"
  />
  <p>
    One Punch Man
  </p>
</div>
`;
